tests/mocha/trashcan_test.js

```diff
@@ -0,0 +1,561 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2019 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+suite("Trashcan", function() {
+  var workspace = {
+    addChangeListener: function(func) {
+      this.listener = func;
+    },
+    triggerListener: function(event) {
+      this.listener(event);
+    },
+    options: {
+      maxTrashcanContents: Infinity
+    }
+  };
+  function sendDeleteEvent(xmlString) {
+    var xml = Blockly.Xml.textToDom(xmlString);
+    xml = xml.children[0];
+    var event = {
+      type: Blockly.Events.BLOCK_DELETE,
+      oldXml: xml
+    };
+    workspace.triggerListener(event);
+  }
+
+  setup(function() {
+    this.trashcan = new Blockly.Trashcan(workspace);
+    this.setLidStub = sinon.stub(this.trashcan, 'setLidAngle_');
+  });
+  teardown(function() {
+    this.setLidStub.restore();
+    this.trashcan = null;
+  });
+
+  suite("Events", function() {
+    test("Delete", function() {
+      sendDeleteEvent(
```

Comment:
`sendDeleteEvent` can add the `<xml>` and `</xml>` tags to remove duplication.

Reply:
Fixed.